src/sentry_core.c

```diff
@@ -511,6 +500,60 @@ sentry__prepare_event(const sentry_options_t *options, sentry_value_t event,
     return NULL;
 }
 
+sentry_envelope_t *
+sentry__prepare_transaction(const sentry_options_t *options,
+    sentry_value_t transaction, sentry_uuid_t *event_id)
+{
+    sentry_envelope_t *envelope = NULL;
+
+    bool should_skip = !sentry_value_is_true(
+        sentry_value_get_by_key(transaction, "sampled"));
+    if (should_skip) {
+        SENTRY_DEBUG("throwing away transaction due to sample rate");
+        goto fail;
+    }
+    // Field is superfluous, strip so it doesn't leak into the payload
+    sentry_value_remove_by_key(transaction, "sampled");
+
+    SENTRY_WITH_SCOPE (scope) {
+        SENTRY_TRACE("merging scope into event");
+        // Don't include debugging info
+        sentry_scope_mode_t mode = SENTRY_SCOPE_ALL & ~SENTRY_SCOPE_MODULES
+            & ~SENTRY_SCOPE_STACKTRACES;
+        sentry__scope_apply_to_event(scope, options, transaction, mode);
+    }
+
+    sentry__ensure_event_id(transaction, event_id);
+    envelope = sentry__envelope_new();
+    if (!envelope || !sentry__envelope_add_transaction(envelope, transaction)) {
+        goto fail;
+    }
+
+    SENTRY_TRACE("adding attachments to envelope");
```

Comment:
also related to a decision we still have to make: should we send attachments along with transactions? Probably not for the time being, and maybe turn that on later?

Reply:
i've already got attachments noted down as something to investigate, so it makes sense to remove this for the time being and to investigate this properly later. the docs (spec?) suggest that transactions indeed should have attachments: https://develop.sentry.dev/sdk/envelopes/#attachment